### DISCUSSION TITLE
Feature Request: [try_strptime] for error handling on date cast

### DISCUSSION THREAD

**Discussion by lei0zhou** [duckdb/duckdb#11483] [Ideas]
```SELECT strptime('02/03/199a', '%d/%m/%Y') ```

return null instead of error, I am not sure if there is an existing solution

**Comment by orzom411** [comment]
try_cast is a good idea, works well most of the time.  The rub, when the date isn't cleanly defined.  As a suggestion it would be useful if there was a "try_strptime" that returned a null when it can't do the deed.

```
select
    STRPTIME(vd.d, '%d/%m/%Y') AS parsed_date
  from (
        select 
            v.d
          from ( values( '02/03/199a' ) ) v(d)
         where REGEXP_MATCHES(v.d, '^\d{2}/\d{2}/\d{4}$')
        ) AS vd;
```

The thought here is regexp_matches gives you actual viable data from the pattern or nothing, e.g. no error but now you can detect the fault. 

Hope it helps, and thanks to ChatGPT for an assist on the regular expression. :)

**Reply by lei0zhou** [reply]
Thanks @orzom411 ! I appreciate the effort! 

a further iteration maybe like this, it still won't catch invalid dates as `02/31/2020', but prob good enough.

```sql
select if(REGEXP_MATCHES(v.d, '^(0[1-9]|1[0-2])/(0[1-9]|[12][0-9]|3[01])/((19|20)\d\d)$'), strptime(v.d, '%m/%d/%Y'), null) as parsed_date
from ( values( '02/03/1994' ), ( '02/03/199a' ) ) v(d);
```


an official support for `try_strptime` will be appreciated.
